src/Lean/ToLevel.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2023 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Alex Keizer
+-/
+prelude
+import Lean.Expr
+
+/-! # `ToLevel` class
+This module defines `Lean.ToLevel`, which is the `Lean.Level` analogue to `Lean.ToExpr`.
+-/
+
+namespace Lean
+
+universe w
+
+/-- A class to create `Level` expressions that denote particular universe levels in Lean.
+`Lean.ToLevel.toLevel.{u}` evaluates to a `Lean.Level` term representing `u` -/
+class ToLevel.{u} : Type where
+  /-- A `Level` that represents the universe level `u`. -/
+  toLevel : Level
+  /-- The universe itself. This is only here to avoid the "unused universe parameter" error.
+    We'll remove this field once https://github.com/leanprover/lean4/issues/2116 gets fixed.
+  -/
+  univ : ∃ x, x = PUnit.unit.{u} := ⟨_, rfl⟩
```

Comment:
It really doesn't matter, but some alternative encodings if you want:
```suggestion
  univ : PUnit.{u} → True := fun _ => trivial
```
```suggestion
  univ : ∀ x : PUnit.{u}, x = x := fun _ => rfl
```

Or, for fun, one that doesn't make use of anything but the basic logic:
```suggestion
  univ : ∀ (_ : Sort u) (p : Prop), p → p := fun _ _ h => h
```

Reply:
I went with a slightly shorter variation of the last suggestion:  `univ : ∀ (_ : Sort u), True := fun _ => trivial`